src/java-runtime/java/mono/android/Runtime.java

```diff
@@ -16,7 +16,7 @@ private Runtime ()
 	public static native void register (String managedType, java.lang.Class nativeClass, String methods);
 	public static native void notifyTimeZoneChanged ();
 	public static native int createNewContext (String[] runtimeApks, String[] assemblies, ClassLoader loader);
-	public static native int createNewContextWithData (String[] runtimeApks, String[] assemblies, byte[][] assembliesBytes, ClassLoader loader, boolean forcePreloadAssemblies);
+	public static native int createNewContextWithData (String[] runtimeApks, String[] assemblies, byte[][] assembliesBytes, String[] assembliesPaths, ClassLoader loader, boolean forcePreloadAssemblies);
```

Comment:
As with [last time](https://github.com/xamarin/xamarin-android/pull/3365#discussion_r305070762), ABI breakage like this will probably break the Designer.

I also think it's time that we started introducing aspects of https://github.com/xamarin/designer/issues/1597

Does `createNewContextWithData()` *need* to be "complete"?  Or can it "mutate" an existing context?

```java
public class DesignerRuntime {
    private DesignerRuntime() {
    }

    public static native int createNewContext (String[] runtimeApks, String[] assemblies, ClassLoader loader);
    public static void loadAssemblies(int contextID, String[] assembliesNames, byte[][] assembliesBytes, boolean forcePreloadAssemblies);
    public static void loadAssemblies(int contextID, String[] assembliesNames, String[] assembliesPaths, boolean forcePreloadAssemblies);

    public static native void switchToContext (int contextID);

    public static native void destroyContexts (int[] contextIDs);
}
```

Reply:
Fortunately it won't! The designer change that uses the new API is still in PR mode so it's fine to continue iterating on that specific method (the current master version continues using `createNewContext`).

That method needs to be complete because we still require loading all the assemblies up-front when the context is created (adding that last `forcePreloadAssemblies` parameter is a way to disable that potentially in a future when we can do it)